tests/utils/avalanche_validator_registry.go

```diff
@@ -0,0 +1,230 @@
+}
+
+func RegisterAvalancheValidatorSet(
+	ctx context.Context,
+	fundedKey *ecdsa.PrivateKey,
+	chainID *big.Int,
+	client ethclient.Client,
+	validatorSetRegistry *avalanchevalidatorsetregistry.AvalancheValidatorSetRegistry,
+	avalancheNetworkID uint32,
+	avalanchePChainBlockchainID ids.ID,
+	avalancheSubnetID ids.ID,
+	avalancheL1BlockchainID ids.ID,
+	pChainClient *platformvm.Client,
+	signatureAggregator *SignatureAggregator,
+) *big.Int {
+	opts, err := bind.NewKeyedTransactorWithChainID(fundedKey, chainID)
+	Expect(err).Should(BeNil())
+
+	// Get the validator set information to create the serialized pre-image.
+	pChainHeight, err := pChainClient.GetHeight(ctx)
+	Expect(err).Should(BeNil())
+	pChainBlockBytes, err := pChainClient.GetBlockByHeight(ctx, pChainHeight)
+	Expect(err).Should(BeNil())
+
+	// Parse the block
+	pChainBlock, err := block.Parse(block.Codec, pChainBlockBytes)
+	Expect(err).Should(BeNil())
+
+	// Confirm that the block is a Banff block
+	banffBlock, ok := pChainBlock.(block.BanffBlock)
+	Expect(ok).Should(BeTrue())
+
+	// Get the validators from the block height
+	rawValidators, err := pChainClient.GetValidatorsAt(ctx, avalancheSubnetID, api.Height(pChainHeight))
+	Expect(err).Should(BeNil())
+
+	canonicalValidatorSet, err := warp.FlattenValidatorSet(rawValidators)
+	Expect(err).Should(BeNil())
+
+	log.Info("Canonical validator set", "numValidators", len(canonicalValidatorSet.Validators), "totalWeight", canonicalValidatorSet.TotalWeight)
+
+	// Create the unsigned message including the validator set hash in the payload.
+	validators := make([]*message.Validator, len(canonicalValidatorSet.Validators))
+	for i, validator := range canonicalValidatorSet.Validators {
+		validators[i] = &message.Validator{
+			UncompressedPublicKeyBytes: [96]byte(validator.PublicKey.Serialize()),
+			Weight:                     validator.Weight,
+		}
+	}
+	validatorsBytes, err := message.Codec.Marshal(message.CodecVersion, validators)
+	Expect(err).Should(BeNil())
+
+	validatorSetHash := sha256.Sum256(validatorsBytes)
+	validaterSetStatePayload, err := message.NewValidatorSetState(
+		avalancheL1BlockchainID,
+		pChainHeight,
+		uint64(banffBlock.Timestamp().Unix()),
+		validatorSetHash,
+	)
+	Expect(err).Should(BeNil())
+
+	addressCallPayload, err := payload.NewAddressedCall([]byte{}, validaterSetStatePayload.Bytes())
+	Expect(err).Should(BeNil())
+
+	// Create the unsigned message
+	unsignedMessage, err := warp.NewUnsignedMessage(avalancheNetworkID, avalanchePChainBlockchainID, addressCallPayload.Bytes())
+	Expect(err).Should(BeNil())
+
+	// create the ICM message suitable for Ethereum
+	icmMessage := PrepareICMMessageForEthereum(
+		avalancheNetworkID,
+		avalanchePChainBlockchainID,
+		avalancheSubnetID,
+		avalanchePChainBlockchainID,
+		signatureAggregator,
+		unsignedMessage,
+	)
+	log.Info("DEBUG: Validator Set Details:",
+		"validatorsBytesLength", len(validatorsBytes),
+		"validatorsBytes", hex.EncodeToString(validatorsBytes),
+	)
```

Comment:
Not sure if this is an `info` because you don't want to see the other debug logs, or you're not able to set the logs to debug or there's a problem where we're not able to set it?